applications/plugins/MultiThreading/src/MeanComputation.inl

```diff
@@ -0,0 +1,179 @@
+                        input->setValue(*mObj->x.beginEdit());
+                        input->setParent(&mObj->x, std::string("to"));
+                        input->setReadOnly(true);
+                        input->setDirtyValue();
+
+                        _inputs.push_back(input);
+                    }
+                }
+
+                // get the lowest size
+                _resultSize = MechObjSizeMap.begin()->first;
+                d_result.beginEdit()->resize(_resultSize);
+
+                compute();
+            }
+
+            template <class DataTypes>
+            void MeanComputation<DataTypes>::bwdInit()
+            {
+                //const size_t nbInputs = _inputs.size();
+                //if (nbInputs == 0)
+                //    return;
+
+                //// check for input vector size different
+                //const size_t vecSize = _inputs[0]->beginEdit()->size();
+                //size_t minVecSize = vecSize;
+
+                //for (size_t i = 1; i<nbInputs; ++i)
+                //{
+                //    const size_t i_size = _inputs[i]->beginEdit()->size();
+
+                //    if (vecSize != i_size)
+                //    {
+                //        minVecSize = (vecSize < i_size) ? vecSize : i_size;
+                //    }
+                //}
+
+                //_resultSize = minVecSize;
+
+                //// set result to 0
+                //d_result.beginEdit()->resize(_resultSize, Coord());
+
+                //compute();
+            }
+
+            template <class DataTypes>
+            void MeanComputation<DataTypes>::reinit()
+            {
+                compute();
+            }
+
+
+            template <class DataTypes>
+            void MeanComputation<DataTypes>::handleEvent(core::objectmodel::Event* event)
+            {
+                if (dynamic_cast<simulation::AnimateBeginEvent*>(event) != NULL)
+                {
+                    compute();
+                }
+            }
+
+           
+            template <class DataTypes>
+            void MeanComputation<DataTypes>::compute()
+            {
+                VecCoord& result = *d_result.beginEdit();
+
+                const size_t nbInputs = _inputs.size();
+                if (nbInputs == 0)
+                    return;
+
+                const double invNbInputs = 1.0 / nbInputs;
+
+                // accumulate all the input elems in result
+                for (int j = 0; j<nbInputs; ++j)
+                {
+                    const VecCoord& pos = *_inputs[j]->beginEdit();
+
+                    for (int i = 0; i<_resultSize; ++i)
+                    {
+                        result[i] += pos[i];
+                    }
+                }
+
+                for (int i = 0; i<_resultSize; ++i)
+                {
+                    result[i] *= invNbInputs;
+                }    
+                
+                d_result.endEdit();
+                d_result.setDirtyValue();
```

Comment:
Normally it is not needed to setDirtyValue. 
Begin/endEdit automatically set dirty to true.

Could you test it without the line and tell me if it breaks something. For my understanding of Sofa Data.

Reply:
Removed the d_result.setDirtyValue() and everything seems to be fine.